torch/_dynamo/output_graph.py

```diff
@@ -731,7 +730,9 @@ def append_prefix_insts():
             self.add_output_instructions(output + pass2.get_instructions())
             codegen = pass2
 
-        if config.numpy_ndarray_as_tensor:
+        if config.numpy_ndarray_as_tensor and (
+            reason is None or not reason.graph_break
```

Comment:
I was not expecting this change. How come?

Reply:
So if we need to compile the graph before return, we need to convert all the torch_np.ndarray into numpy.ndarray. However if we compile the graph because of a graph break, we actually want the value to keep being torch_np.ndarray.

Reply:
OK, this is actually the same thing as https://github.com/pytorch/pytorch/pull/100839#discussion_r1189301724

My take here is that if there is a graph break, you MUST turn the internal torch_np.ndarray into numpy.ndarray. You don't get to skip it "just because there's a graph break". What if we never successfully trace after the graph break?

Reply:
I see that makes sense. Then if we start to trace again, I need to convert those into torch_np.ndarray again? Where do we do that?

Reply:
Ah torch_np APIs take numpy.ndarray. Then we don't have to do any conversion to torch_np.ndarray.

Reply:
Ah, yes, so one piece that might be missing right now, is if we start compiling and there are some numpy.ndarray, you want to turn these into NumpyNdarrayVariable so we can compile them. The E2E then looks like "make tensors out of the ndarrays, do PyTorch compute, then put it back into ndarray)

Reply:
ok so there's a glitch on passing numpy.ndarray directly into torch_np APIs and from the discussion here it seems not recommended: https://github.com/Quansight-Labs/numpy_pytorch_interop/issues/136

If we can convert them into tensor we don't have problem passing them to torch_np, but that's not an option though since we have to keep these values as numpy.ndarray.

Reply:
No I don't understand. When I compile a function that takes ndarray as inputs, I expect the entry code into the compiled graph to convert the ndarrays into tensors, so then we can just run our compiled tensor compute on it. This should be O(1) most of the time. What is happening in your PR today?

Reply:
Ah I think I know what you mean. Let me update the PR.

Reply:
Ok updated. Basically we need to convert np.ndarray to torch_np.ndarray when we take them as inputs. Added this logic in builder.py.